src/main/java/prefs/Preferences.java

```diff
@@ -10,160 +16,296 @@
     public void removeBoard(String name) {
-        global.removeBoard(name);
+        sessionConfig.removeBoard(name);
+        save();
     }
 
+    /**
+     * Sets the last open board
+     * @param board The name of the last open board.
+     */
     public void setLastOpenBoard(String board) {
-        global.setLastOpenBoard(board);
+        sessionConfig.setLastOpenBoard(board);
+        save();
     }
 
+    /**
+     * Retrieves the name of the last open board.
+     * @return An Optional of a board name
+     */
     public Optional<String> getLastOpenBoard() {
-        return global.getLastOpenBoard();
+        return sessionConfig.getLastOpenBoard();
     }
 
+    /**
+     * Switches the board to the next one. Cycles through the boards one at a time.
+     * @return The new board selected
+     */
     public Optional<String> switchBoard() {
         if (getLastOpenBoard().isPresent() && getAllBoards().size() > 1) {
             List<String> boardNames = getAllBoardNames();
             int lastBoard = boardNames.indexOf(getLastOpenBoard().get());
             int index = (lastBoard + 1) % boardNames.size();
-
+            
             setLastOpenBoard(boardNames.get(index));
         }
-
         return getLastOpenBoard();
     }
 
+    /**
+     * Clears the last open board
+     */
     public void clearLastOpenBoard() {
-        global.clearLastOpenBoard();
+        sessionConfig.clearLastOpenBoard();
+        save();
     }
 
-    public List<PanelInfo> getBoardPanels(String board) {
-        return global.getBoardPanels(board);
+    /**
+     * Retrieves the panels of a board.
+     * @param boardName The name of the board to retrieve the panels from
+     * @return A list of panel infos from that board
+     */
+    public List<PanelInfo> getBoardPanels(String boardName) {
+        return sessionConfig.getBoardPanels(boardName);
     }
 
+    /**
+     * Clears all boards.
+     */
     public void clearAllBoards() {
-        global.clearAllBoards();
+        sessionConfig.clearAllBoards();
+        save();
     }
 
     /**
-     * Session configuration
+     * Sets the last viewed repo to the specified repo
+     * @param repository The last viewed repo
      */
-
     public void setLastViewedRepository(String repository) {
-        global.setLastViewedRepository(repository);
+        sessionConfig.setLastViewedRepository(repository);
+        save();
     }
 
+    /**
+     * Retrieves the last viewed repo.
+     * @return An Optional of the last viewed repo.
+     */
     public Optional<RepositoryId> getLastViewedRepository() {
-        if (global.getLastViewedRepository().isEmpty()) {
+        if (sessionConfig.getLastViewedRepository().isEmpty()) {
             return Optional.empty();
         } else {
-            RepositoryId repositoryId = RepositoryId.createFromId(global.getLastViewedRepository());
+            RepositoryId repositoryId = RepositoryId.createFromId(sessionConfig.getLastViewedRepository());
```

Comment:
We can treat the `if` case as guard clause so this one can be taken out of `else`.